### PR TITLE
Hamster Pet (GhostRoleMob)

### PR DESCRIPTION
**Pull request by brainfood1183**
Untitled-120

This pr adds a hamster pet and hamster cage.
The hamster pet is called Hamlet, he lives in the station bridge (although that is up to each individual mapper). Hamlet is a ghostrole mob from the start of the round, he lives in his cage in the bridge and cannot leave his cage unless someone opens the cage for him. He then functions similar to a mouse and can move around the station and under doors. Hamlet cannot communicate except for squeaks and "piep". Hamlet can attack for 2 damage (a lot less than remi). Hamlet has more health than a mouse and is slower than a mouse. Hamlet can be picked up by the station crew just like a mouse but cannot be worn on a crewmembers head. Hamlet is not going to be an easy ghostrole to play due to the fact crewmembers can mess with him just like a mouse.

Hamsters can be ordered from cargo at a cost of 2800. Quite expensive but each hamster is a ghostrole mob similar to Hamlet (but not as robust). If cargo wishes they can buy in lots of hamsters and have some fun in cargo.

- Hamlet pet NPC
- Hamster NPC
- Hamster Cage crate
- Hamster Cage livestock ordered from cargo request
- Hamlet spawner (for spawning in hamlet and needs to be placed in a map by a mapper if that mapper wants hamlet to spawn).

Going into a little more detail here about the stats of the hamsters and what they can and cannot do.

Hamlet

- same health as a rat servant and the same health loss before crit.
-  can attack for 2 damage with his bite.
- can pull objects about
- can be picked up by crewmembers
- can speak albeit using the mouse accent (he pieps).
- can be killed by mousetraps however it takes 2 traps to kill hamlet. 1 trap will leave hamlet on very low health.
- starts inside a rodent cage, cannot move whilst in the rodent cage or escape it. Can speak (using mouse speak pieps) to attract attention, if a crewmember opens the cage then hamlet can roam freely.
- hamlet can move under tables and doors.
- hamlet is a starting pet and would need a spawn adding to the station map to spawn, i recommend this be inside the bridge area.
- hamlet has a 100% chance of being a ghost role (i think this is balanced by hamlet being a slightly more robust mouse and still quite fragile).

Hamster

- same health as a rat servant and the same health loss before crit.
- can attack for 2 damage with bite.
- cannot pull objects
- can be picked up by crewmembers
- can speak albeit using the mouse accent (pieps).
- can be killed by mousetraps however it takes 2 traps to kill hamster. 1 trap will leave hamster on very low health.
- starts inside a rodent cage, cannot move whilst in the rodent cage or escape it. Can speak (using mouse speak pieps) to attract attention, if a crewmember opens the cage then hamster can roam freely.
- hamster can move under tables and doors.
- hamsters can be ordered from cargo for 2800 space bucks and arrive in a rodent cage.
- hamsters ordered via cargo have a 100% chance of being a ghost role, this is something i am not sure on but the high price to order a crate does currently reflect this.

changes made since initial PR

- Hamlet now has his own ghost role name and description (mirrorcult requested change).
- Fixed a typo in the locale for petting chance (mirrorcult requested change).
- Hamsters and Hamlet now have density 120 and can survive a single mousetrap.
- Hamlet can pull objects around (hamlet does not have hands though).

### CODE REVIEW

File: Resources/Locale/en-US/interaction/interaction-popup-component.ftl [hunk 1]
```diff
@@ -32,6 +33,7 @@ petting-failure-possum = You reach out to pet {THE($target)}, but are met with h
 petting-failure-sloth = You reach out to pet {THE($target)}, but {SUBJECT($target)} somehow dodge with ludicrous speed!
 petting-failure-holo = You reach out to pet {THE($target)}, but {POSS-ADJ($target)} spikes almost impale your hand!
 petting-failure-dragon = You raise your hand, but as {THE($target)} roars, you decide you'd rather not be toasty carp food.
+petting-failure-hamster = You reach out to pet {THE($target)}, but {SUBJECT($target)} attempts to bite your finger, you quickly recoil your hand.
```
Comment (mirrorcult) [inline Resources/Locale/en-US/interaction/interaction-popup-component.ftl:36]:
```suggestion
petting-failure-hamster = You reach out to pet {THE($target)}, but {SUBJECT($target)} attempts to bite your finger, so you quickly recoil your hand.
```

File: Resources/Locale/en-US/ghost/roles/ghost-role-component.ftl [hunk 2]
```diff
@@ -9,4 +9,7 @@ ghost-role-information-mouse-description = A hungry and mischievous mouse.
 ghost-role-information-giant-spider-name = Giant spider
 ghost-role-information-giant-spider-description = Wreak havoc on the station's inhabitants!
 
-ghost-role-information-cognizine-description = Made conscious with the magic of cognizine.
\ No newline at end of file
+ghost-role-information-cognizine-description = Made conscious with the magic of cognizine.
+
+ghost-role-information-hamster-name = Hamster
```
Comment (mirrorcult) [inline Resources/Locale/en-US/ghost/roles/ghost-role-component.ftl:14]: shouldnt hamlet have his own ghost role info?

**Review by mirrorcult — CHANGES_REQUESTED** [review]

**Comment by mirrorcult** [comment]
love these sprites

File: Resources/Locale/en-US/ghost/roles/ghost-role-component.ftl (hunk 2 above)
Comment (brainfood1183) [reply Resources/Locale/en-US/ghost/roles/ghost-role-component.ftl:14]: I musat have missed that, will take a look tomorrow

File: Resources/Locale/en-US/interaction/interaction-popup-component.ftl (hunk 1 above)
Comment (brainfood1183) [reply Resources/Locale/en-US/interaction/interaction-popup-component.ftl:36]: i will fix it although currently i do not think you can pet mice or hamster due to fact that when u click on a mouse u pick it up.

File: Resources/Locale/en-US/ghost/roles/ghost-role-component.ftl (hunk 2 above)
Comment (brainfood1183) [reply Resources/Locale/en-US/ghost/roles/ghost-role-component.ftl:14]: hamlet now has his own unique ghost role description and name.

File: Resources/Locale/en-US/interaction/interaction-popup-component.ftl (hunk 1 above)
Comment (brainfood1183) [reply Resources/Locale/en-US/interaction/interaction-popup-component.ftl]: i changed this line to the following.

"petting-failure-hamster = You reach out to pet {THE($target)}, but {SUBJECT($target)} attempts to bite your finger and only your quick reflexes save you from an almost fatal injury."

Would love to know your opinion on the change or if you have a suggestion for something better.

Comment (mirrorcult) [reply Resources/Locale/en-US/interaction/interaction-popup-component.ftl]: sounds good

**Review by mirrorcult — APPROVED** [review]

**Pull request merged**